gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/component/GbDataTable.java

```diff
@@ -0,0 +1,49 @@
+package org.sakaiproject.gradebookng.tool.component;
+
+import org.apache.wicket.extensions.markup.html.repeater.data.table.AbstractColumn;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.DataTable;
+import org.apache.wicket.extensions.markup.html.repeater.data.table.IColumn;
+import org.apache.wicket.markup.ComponentTag;
+import org.apache.wicket.markup.repeater.data.IDataProvider;
+import org.apache.wicket.markup.repeater.Item;
+import org.apache.wicket.model.IModel;
+
+import java.util.List;
+import java.util.Map;
+
+public class GbDataTable<T, S> extends DataTable<T, S> {
+
+	IModel<Map<String, Object>> gradebookData;
+
```

Comment:
Yep, I'm happy to leave them as passed in via the model as there'll be no extra service calls.  I was expecting we'd get rid of `GbDataTable` and move `newCellItem` override line on in GradebookPage.java like  `new DataTable {@Override... newCellItem...}`.  The only thing holding me back from doing that was the generics and compilation errors when I tried to get it working.